### PR TITLE
mgmt privatedns avoid unrequired update

### PR DESCRIPTION
**Pull request by xseeseesee**

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java [hunk 1]
```diff
@@ -454,4 +473,17 @@ public Update withETagCheck(String etagValue) {
                 }
             });
     }
+
+    private String serializeTags(Map<String, String> tags) {
+        try {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java:478]: Can you just do a new HashMap? It does not really need this kind of JSON, at least not now.

File: sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java [hunk 2]
```diff
@@ -394,6 +411,7 @@ public Update withETagCheck(String etagValue) {
             .map(innerToFluentMap(this))
             .map(privateDnsZone -> {
                 etagState.clear();
+                tagsCopy = serializeTags(privateDnsZone.tags());
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java:414]: I would suggest you do this on `update`. Else I won't know how many places this line is needed (e.g. if I add a `updateResourceAsync` method later, how do I know I need to do this?).

File: sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java (hunk 1 above)
Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java:478]: Do you mean create a deep copy or have add/remove maps?

File: sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java (hunk 2 above)
Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java:414]: Same concern would apply to ETagState too?

File: sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java:478]: There is no deep copy for a HashMap, you just create a new one via constructor.

Its content is String, which is immutable.

File: sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java:414]: ETagState might be a bit different? As it is used both in create or update.

But the `tagsCopy` is only for update flow, hence it just need to be done in the entry point of update flow, which is `update` method.

**Pull request closed without merging**